packages/kbn-esql-validation-autocomplete/src/shared/helpers.ts

```diff
@@ -289,7 +298,7 @@ const arrayToSingularMap: Map<FunctionParameterType, FunctionParameterType> = ne
  * Given an array type for example `string[]` it will return `string`
  */
 export function extractSingularType(type: FunctionParameterType): FunctionParameterType {
-  return arrayToSingularMap.get(type) ?? type;
+  return isArrayType(type) ? arrayToSingularMap.get(type)! : type;
```

Comment:
Looking at this again, will there ever be a case where we can not remove the last `[]` in the text to get the singular type? Do we need to look up a map for it?

Reply:
We could always just remove the `[]` through string manipulation. I like the map because it makes it clear which types can be arrays. I guess this whole array types thing makes me nervous because I don't have a good handle yet on its purpose. So, I like being a bit more explicit.

Open to opinions though.